### ISSUE TITLE
Sample skill doesn't work.

### ISSUE THREAD

**Issue by DancinDirk** [alexa/skill-sample-nodejs-audio-player#85]
Hello Everyone.

I've made several attempts to get this multiple stream sample code up and running with no luck. I've followed the steps in the multiple streams readme exactly, and even after working through the bad json in .ask/config I'm getting errors.

I have done all steps through ask deploy and assigning the IAM roles. When I do the ask simulate step to test the install I get the following.

`$ ask simulate -l en-US -t "alexa, open audio player"
✓ Simulation created for simulation id: 9a4201c8-XXXX-424b-XXXX-b95cc4b78f59
◡ Waiting for simulation response{
  "id": "9a4201c8-XXXX-424b-XXXX-b95cc4b78f59",
  "status": "SUCCESSFUL",
  "result": {
    "skillExecutionInfo": {
      "invocationRequest": {
        "endpoint": "arn:aws:lambda:us-east-1:XXXX:function:ask-custom-Multi_Stream_Audio_Player",
        "body": {
          "version": "1.0",
          "session": {
            "new": true,
            "sessionId": "amzn1.echo-api.session.b5e7ee44-XXXX-4bdc-XXXX-a90102c34271",
            "application": {
              "applicationId": "amzn1.ask.skill.c62a3314-XXXX-440d-XXXX-fcc7eb1d8fe9"
            },
            "user": {
              "userId": "amzn1.ask.account.AFFG7XZAQJ2UDTWL77DLE3AGPTGFHEMSEPRZ6O7XPH53WMAJTEXW4LZ4XXXXLWIWFDA3QLTRGM3CATMOHSQ4T5P5HMYWIDWUUUFYJJ3EXFWCESD3UEZKPHWDGMPICX2LRUF4AWFEE4UNWLSQB2X4RZ46ACRLPQSMXDFRJCNPRLUKFRJXX2F76V4PNG5FCKERKR3ZY"
            }
          },
          "context": {
            "Display": {},
            "System": {
              "application": {
                "applicationId": "amzn1.ask.skill.c62a3314-XXXX-440d-XXXX-fcc7eb1d8fe9"
              },
              "user": {
                "userId": "amzn1.ask.account.AFFG7XZAQJ2UDTWL77DLE3AGPTGFHEMSEPRZ6O7XPH53WMAJTEXW4LZ4XXXXLWIWFDA3QLTRGM3CATMOHSQ4T5P5HMYWIDWUUUFYJJ3EXFWCESD3UEZKPHWDGMPICX2LRUF4AWFEE4UNWLSQB2X4RZ46ACRLPQSMXDFRJCNPRLUKFRJXX2F76V4PNG5FCKERKR3ZY"
              },
              "device": {
                "deviceId": "amzn1.ask.device.AH65GKUL4QNANDG5V5EDEOPSMCL3ODYROKCXNBBYDTKCQWBQZLF3LGXXXXBZ4IUNUZHTP76KMSJDNUZL6HM2NSRQ43YY4UFSI3ZE6CH7GZKW7X55VORDRGYZ4VJP6BFALAY44PSHFZALHHXLNJCSQ",
                "supportedInterfaces": {
                  "AudioPlayer": {},
                  "Display": {
                    "templateVersion": "1.0",
                    "markupVersion": "1.0"
                  }
                }
              },
              "apiEndpoint": "https://api.amazonalexa.com",
              "apiAccessToken": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6IjEifQ.eyJhdWQiOiJodHRwczovL2FwaS5hbWF6b25hbGV4YS5jb20iLCJpc3MiOiJBbGV4YVNraWxsS2l0Iiwic3ViIjoiYW16bjEuYXNrLnNraWxsLmM2MmEzMzE0LTViM2QtNDQwZC1hOGYwLWZjYzdlYjFkOGZlOSIsImV4cCI6MTUyMTgzNzY3NCwiaWF0IjoxNTIxODM0MDc0LCJuYmYiOjE1MjE4MzQwNzQsInByaXZhdGVDbGFpbXMiOnsiY29uc2VudFRva2VuIjpudWxsLCJkZXZpY2VJZCI6ImFtem4xLmFzay5kZXZpY2UuQUg2NUdLVUw0UU5BTkRHNVY1RURFT1BTTUNMM09EWVJPS0NYTkJCWURUS0NRV0JRWkxGM0xHSkJFN0JIWlYzNEVCWjRJVU5VWkhUUDc2S01TSkROVXXXXMk5TUlE0M1lZNFVGU0kzWkU2Q0g3R1pLVzdYNTVWT1JEUkdZWjRWSlA2QkZBTEFZNDRQU0hGWkFMSEhYTE5KQ1NRIiwidXNlcklkIjoiYW16bjEuYXNrLmFjY291bnQuQUZGRzdYWkFRSjJVRFRXTDc3RExFM0FHUFRHRkhFTVNFUFJaNk83WFBINTNXTUFKVEVYVzRMWjRTNkZKNTREQlZVQVBHVUxXSVdGREEzUUxUUkdNM0NBVE1PSFNRNFQ1UDVITVlXSURXVVVVRllKSjNFWEZXQ0VTRDNVRVpLUEhXREdNUElDWDJMUlVGNEFXRkVFNFVOV0xTUUIyWDRSWjQ2QUNSTFBRU01YREZSSkNOUFJMVUtGUkpYWDJGNzZWNFBORzVGQ0tFUktSM1pZIn19.jnW1ATDsY0ozMTZVwghVE0LtcmzWrp-uF062trLXnizFyK0zE41EaUR-G9Uw2Py_-qGpJcsJKKAjytRME6LfrRUmk5ftoXKweLRonbaCQKAf0OG3LumZnR-c5Tr4OWGPb_unQxp-Nflz6u8L9hJK-ETV7ilByuqDie8Sg1eXOXJhoeuvmr-kqiQgIUrqeD8_JAOuXOnqM1AfzQNFACAynZ3IKSeR40HdqWJ2zipn1HR1x0Q6pEJZOg0zFw4tXB7y0ysNKSnDKdZn9VXkfPvhIwVTGD3YIKAyYKofjGlI2CuODkY-TVHF4b6ZBd1fZwM2YbOtL5HvWyQJcWDrLVe8pA"
            }
          },
          "request": {
            "type": "LaunchRequest",
            "requestId": "amzn1.echo-api.request.9f324e17-XXXX-4063-XXXX-98b50974267e",
            "timestamp": "2018-03-23T19:41:14Z",
            "locale": "en-US"
          }
        }
      },
      "metrics": {
        "skillExecutionTimeInMilliseconds": 1996
      }
    },
    "error": {
      "message": "Request to skill endpoint resulted in an error."
    }
  }
}
`

Has anyone got the steps to get this example code working?
Thanks in advance!

**Comment by sebsto** [comment]
Hello Dirk,

Sorry to read you have problems with the sample code.  Can you copy the result of the console.log() statement in the lambda function to show the full response sent from lambda to Alexa ?

https://github.com/alexa/skill-sample-nodejs-audio-player/blob/mainline/multiple-streams/lambda/src/index.js#L25

You can access your lambda log files from AWS Cloudwatch logs.  

Thanks

**Comment by DancinDirk** [comment]
Thanks for pointing me in the right direction! After looking at the Lambda logs I was able to figure out the problem. I also think looking at it with a fresh set of eyes a couple days later helped.

If you followed the readme.md exactly it did not work. I have submitted a pull request with a change to the readme that adds a missing step necessary to deploy successfully. Short version, it's the step of copying the newly created skill ID into the constants.js file and deploying a second time. Probably obvious to experienced lambda/alexa devs but the change should help out devs who are new to lambda/alexa.

Thanks again!
